.github/actions/learning-path-staleness-check/index.js

```diff
@@ -0,0 +1,288 @@
+    const link = learningPathContents.substring(startOfLink, endOfLink);
+
+    if (excludeLinksArray.some(excludeLink => link.toLowerCase().includes(excludeLink))) { continue; }
+
+    const pathStartIndex = link.indexOf(sourceDirectoryName);
+    if (pathStartIndex === -1) { continue }
+
+    if (!link.includes(oldHash))
+    {
+      UpdateOutOfSync(link, learningPathFile);
+      continue
+    }
+
+    const linePrefixIndex = link.indexOf(linePrefix);
+    const linkHasLineNumber = linePrefixIndex !== -1;
+    const pathEndIndex = linkHasLineNumber ? linePrefixIndex : endOfLink;
+
+    // Check if the file being referenced by the link is one of the modified files in the PR
+    const linkFilePath = link.substring(pathStartIndex, pathEndIndex);
+    if (modifiedPRFiles.includes(linkFilePath))
+    {
+      const fileName = linkFilePath.substring(linkFilePath.lastIndexOf('/') + 1);
+
+      UpdateModifiedFiles(fileName, linkHasLineNumber ? StripLineNumber(link, linePrefixIndex) : link, learningPathFile);
+
+      // This is the line number in the learning path file that contains the link - not the #L line number in the link itself
+      const learningPathLineNumber = learningPathContents.substring(0, startOfLink).split("\n").length;
+
+      var headContent = GetContent(linkFilePath)
+      if (!headContent) {
+        UpdateManuallyReview(fileName, link, learningPathFile, learningPathLineNumber);
+        continue
+      }
+      const headContentLines = headContent.toString().split("\n");
+
+      if (!linkHasLineNumber) { continue; }
+      const oldLineNumber = Number(link.substring(linePrefixIndex + linePrefix.length, link.length));
+
+      var prevContent = GetContent(prevPathPrefix + linkFilePath)
+      if (!prevContent) { continue; }
+      const prevContentLines = prevContent.toString().split("\n");
+
+      if (prevContentLines.length < oldLineNumber)
+      {
+        UpdateManuallyReview(fileName, link, learningPathFile, learningPathLineNumber, oldLineNumber);
+      }
+      else if (headContentLines.length < oldLineNumber || prevContentLines[oldLineNumber - 1].trim() !== headContentLines[oldLineNumber - 1].trim())
+      {
+        const newLineNumberLast = headContentLines.lastIndexOf(prevContentLines[oldLineNumber - 1]) + 1;
+        const newLineNumberFirst = headContentLines.indexOf(prevContentLines[oldLineNumber - 1]) + 1;
+
+        if (newLineNumberLast !== newLineNumberFirst) // Multiple matches found in the file
+        {
+          UpdateManuallyReview(fileName, link, learningPathFile, learningPathLineNumber, oldLineNumber);
+        }
+        else
+        {
+          let updatedLink = StripLineNumber(link.replace(oldHash, newHash), linePrefixIndex) + linePrefix + newLineNumberFirst;
+          UpdateSuggestions(fileName, link, updatedLink, learningPathFile, learningPathLineNumber, oldLineNumber, newLineNumberFirst);
+        }
+      }
+    }
+  }
+}
+
+const main = async () => {
```

Comment:
Similar feedback to `ValidateLinks`, and can also be addressed in a seperate PR.

This method is doing too much so it's hard to follow. Ideally a method is doing one thing with clear input and output. In the case of `main` or other methods that orchestrate a series of actions, only having them contain flow-control logic and not doing implementation themselves keeps things easier to follow and often less prone to bugs that'd get missed in a review.

Reply:
Though the whole thing could use some refactoring (as mentioned in another comment), this in particular was a side effect of compressing the two actions into one, and I just put it here for simplicity when I did that.